app/models/user.rb

```diff
@@ -138,6 +138,31 @@ def is_admin
     is_global_admin || community_user&.is_admin || false
   end
 
+  # Used by network profile: does this user have a profile on that other comm?
+  def has_profile_on(community_id)
+    cu = community_users.where(community_id: community_id).first
+    !cu&.user_id.nil?
+  end
+
+  def reputation_on(community_id)
+    cu = community_users.where(community_id: community_id).first
+    cu&.reputation
```

Comment:
Optional: same as with the `post_count_on` and `reputation_on` suggestions, let's default to `1` here in case `cu` fails the safe navigation check.